### PR TITLE
Lib 305 add analytics

### PR DESCRIPTION
**Pull request by jjwill**

### CODE REVIEW

File: src/plugin-segment/analytics.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
+
+export default (function () {
+  if (!ExecutionEnvironment.canUseDOM) {
+    return null;
+  }
+
+  return {
+    onRouteUpdate() {
```
Comment (jjwill) [inline src/plugin-segment/analytics.js:9]: Required for SPAs

File: src/plugin-segment/index.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  if (!productionKey) {
+    throw new Error(
+      'You specified the `segment` object in `customFields` but the `productionKey` field was missing. ' +
+        'Please ensure this is not a mistake.',
+    );
+  }
+
+  const isProd = process.env.NODE_ENV === 'production';
+  const apiKey = isProd ? productionKey : stagingKey;
```
Comment (jjwill) [inline src/plugin-segment/index.js:27]: Allows us to use a staged account

File: src/plugin-segment/analytics.js (hunk 1 above)
Comment (oriooctopus) [reply src/plugin-segment/analytics.js:9]: This looks good... just wondering what's the point of an iife (immedietely invoked function expression) here?

**Comment by oriooctopus** [comment]
Definitely the cleanest way to go!

**Pull request merged**